pkgs/stdenv/darwin/default.nix

```diff
@@ -2269,81 +1118,88 @@ assert bootstrapTools.passthru.isFromBootstrapFiles or false; # sanity check
 
         initialPath = ((import ../generic/common-path.nix) { pkgs = prevStage; });
 
-        extraNativeBuildInputs =
-          lib.optionals localSystem.isAarch64 [ prevStage.updateAutotoolsGnuConfigScriptsHook ]
-          ++ [ prevStage.darwin.apple_sdk.sdkRoot ];
+        extraNativeBuildInputs = lib.optionals localSystem.isAarch64 [
+          prevStage.updateAutotoolsGnuConfigScriptsHook
+        ];
```

Comment:
I don’t know about making them unconditional. That’s outside the scope of this change.

> More general remark: does it make sense to migrate to buildPlatform/hostPlatform rather than the confusing localSystem stuff?

That would be a change that should be made to all the stdenv definitions.